### PR TITLE
fix ERROR_PIPE_CONNECTED(535) for ConnectNamedPipe

### PR DESCRIPTION
**Pull request by fishjam**
PR for [ERROR_PIPE_CONNECTED when ConnectNamedPipe in syelogd if there are lots of clients](https://github.com/microsoft/Detours/issues/241)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/Detours/pull/242)

### CODE REVIEW

File: samples/syelog/syelogd.cpp [hunk 1]
```diff
@@ -180,9 +181,15 @@ PCLIENT CreatePipeConnection(HANDLE hCompletionPort)
         MyErrExit("CreateIoComplietionPort pClient");
     }
 
-    if (!ConnectNamedPipe(hPipe, pClient)) {
-        if (GetLastError() != ERROR_IO_PENDING &&
-            GetLastError() != ERROR_PIPE_LISTENING) {
+    if (!ConnectNamedPipe(hPipe, pClient)) {
+        DWORD dwLastErr = GetLastError();
+        //client already connected while call this fun
```
Comment (bgianfo) [inline samples/syelog/syelogd.cpp:186]: Small nitpick, I think this comment could be more clear, something  like: 
```suggestion
        // Handle race between multiple client connections

```

**Review by bgianfo — CHANGES_REQUESTED** [review]
Happy to merge after addressing the minor comment nits. 

Thanks!

**Review by bgianfo — CHANGES_REQUESTED** [review]
It looks like you might have changed the line endings of this file by accident. The entire file appears to be edited. 

Would you mind undo doing that, it makes any diff extremely hard to read.

**Comment by fishjam** [comment]
sorry, I merged your comment change code on web, seems it change the line ending. 
I will try to undo it, if I failed(or mayb I can not find the way), you can change the code and commit. Anyway, as long as we can solve it.

**Comment by fishjam** [comment]
And I found my previous use the `\r\n` as line ending, it is different from your code. So I change to `\n`.

**Comment by bgianfo** [comment]
Thanks for the patch!

**Pull request merged**